deploy.sh

```diff
@@ -45,54 +47,103 @@ printf "email: $INPUT_EMAIL\npassword: $INPUT_PASSWORD" >>~/.jexia/config.yml
-    exit 0
-elif [ "$STATUS" = "In progress" ]; then
-    echo -e "\e[32mStarting deployment.\e[39m Not waiting for completion"
-    exit 0
-else
-    echo -e "\e[31mDeploy failed. ${ERROR_INFO//\\/}\e[39m"
+OUTPUT=$(eval " ${COMMAND}" 2>&1)
+
+# Get values from Jexia CLIs deploy command by searching for them in the string
+STATUS=$(echo "${OUTPUT}" | grep -oP '(?<=status=").*?(?=")')
+ERROR_INFO=$(echo "${OUTPUT}" | grep -oP '(?<=info=").*?(?=")')
+
+# Check if the status value is returned, if not search the output for know phrases related to errors
+# This process allows us to better handle when an exit code 0 and exit code 1 are used by searching the
+# error for a string if the output is not returned in the desired shell format.
+if [ -z "${STATUS}" ]; then
+
+    # We use a case command to match substrings within the output, we use this to return 'better' errors.
+    case "${OUTPUT}" in
+    *"already in progress"*)
+        # Emulate a known error returned from the deploy command when a deploy is in progress
+        STATUS="Error"
+        ERROR_INFO="a deployment is already running"
+        ;;
+    *"internal server error"*)
+        # Emulate a known error returned from the deploy command when there is an unknown internal server error
+        STATUS="Error"
+        ERROR_INFO="internal server error"
+        ;;
+    *"no matching policy"*)
+        # Emulate a known error returned from the deploy command when the project ID is not found
+        STATUS="Error"
+        ERROR_INFO="incorrect Jexia project ID"
+        serious_error
+        ;;
+    *"incorrect email or password"*)
+        # Emulate a known error returned from the deploy command when authentication fails
+        STATUS="Error"
+        ERROR_INFO="unable to authenticate, incorrect email or password provided"
+        serious_error
+        ;;
+    esac
+
+    # If the user has turned debugging on, we will append the whole command response to the end of ERROR_INFO
+    if [ "$INPUT_DEBUG" = true ]; then
+        # Display the exact command used
+        ERROR_INFO="${ERROR_INFO} ${DARK_COLOR}(${OUTPUT})${RESET_COLOR}"
+    fi
+fi
+
+# Case statement to output the correct message and exit with the correct code
+case "${STATUS}" in
+"Success")
+    echo -e "${SUCCESS_COLOR}Deployed successfully${RESET_COLOR}"
+    ;;
+
+"In progress")
+    echo -e "${SUCCESS_COLOR}Starting deployment.${RESET_COLOR} Not waiting for completion"
+    ;;
+
+"Error")
+    echo -e "${ERROR_COLOR}Deploy failed, ${ERROR_INFO}${RESET_COLOR}"
 
     # This will be useful if they expect to trigger this event frequently where Jexia may not have completed a previous deployment
-    if [ $INPUT_SILENT_FAIL = true ]; then
-        echo -e "\e[31mFailed silently, exit code 0\e[39m"
+    # If it is a serious error, such as the users credentials are incorrect, we will ignore the silent error request.
+    if [ "$INPUT_SILENT_FAIL" = true ] && [ "$SERIOUS_ERROR" = false ]; then
```

Comment:
```suggestion
    if [ "${INPUT_SILENT_FAIL}" = "true" ] && [ "${SERIOUS_ERROR}" = "false" ]; then
```

(for consistency reasons I would use the `${variable}` format here as well, like you did in the rest of the script)